### PR TITLE
Update fastify to v3

### PR DESCRIPTION
**Pull request by frikille**
#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: .pgpass [hunk 1]
```diff
@@ -0,0 +1 @@
+localhost:*:*:postgres:postgres
```
Comment (mcollina) [inline .pgpass:1]: why is this needed?

File: test/initialization.test.js [hunk 2]
```diff
@@ -64,20 +64,20 @@ test('Should not throw if registered within different scopes (with and without n
...
+      connectionString: 'postgres://postgres:postgres@localhost/postgres'
     })
 
     next()
   })
 
   fastify.register(function scopeTwo (instance, opts, next) {
     instance.register(fastifyPostgres, {
-      connectionString: 'postgres://postgres@localhost/postgres',
+      connectionString: 'postgres://postgres:postgres@localhost/postgres',
```
Comment (mcollina) [inline test/initialization.test.js:75]: maybe we can use `.env` to load those up? I think locally it could be different. It would also be different if Docker is used.

Comment (frikille) [reply test/initialization.test.js:75]: It is the same locally as well. I updated to docker commands

File: .pgpass (hunk 1 above)
Comment (frikille) [reply .pgpass:1]: in the ci config we must set a POSTGRESS_PASSWORD value, and when we run the psql [command](https://github.com/fastify/fastify-postgres/pull/60/files#diff-60fe268f71ae7f8409aff40d0a3e51d8R37) it would prompt for a password, but if we provide a `.pgpass` file it will use that to get the password

File: package.json [hunk 3]
```diff
@@ -7,8 +7,8 @@
     "test": "standard && tap -J test/*.test.js",
     "test:report": "standard && tap -J --coverage-report=html test/*.test.js",
     "test:verbose": "standard && tap -J test/*.test.js -Rspec",
-    "load-data": "docker run --rm  -d --link fastify-postgres:postgres postgres:11-alpine psql -h postgres -d postgres -c 'CREATE TABLE users(id serial PRIMARY KEY, username VARCHAR (50) NOT NULL);' -U postgres",
-    "postgres": "docker run --rm -d -p 5432:5432 --name fastify-postgres postgres:11-alpine"
+    "postgres": "docker run -p 5432:5432 --name fastify-postgres -e POSTGRES_PASSWORD=postgres -d postgres",
```
Comment (Eomm) [inline package.json:10]: I would keep the verson
```suggestion
    "postgres": "docker run -p 5432:5432 --name fastify-postgres -e POSTGRES_PASSWORD=postgres -d postgres:11-alpine",
```
